File: scripts/check-sizes.js
#!/usr/bin/env node
import path from 'path';
import { dirname } from 'path';
import { fileURLToPath } from 'url';
import sharp from 'sharp';
import { glob } from 'glob';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

(async () => {
  try {
    const imagesDir = path.resolve(__dirname, '..', 'public', 'img');
    const pattern = path.join(imagesDir, '*.{jpg,JPG,jpeg,JPEG,png,PNG}');

    glob(pattern, async (err, files) => {
      if (err) throw err;
      if (files.length === 0) {
        console.log('No images found.');
        return;
      }

      const metadataList = await Promise.all(
        files.map(file => sharp(file).metadata())
      );

      const dimensions = metadataList.map(meta => `${meta.width}x${meta.height}`);
      const uniqueDims = [...new Set(dimensions)];

      if (uniqueDims.length === 1) {
        console.log(`All ${files.length} images are ${uniqueDims[0]}`);
      } else {
        console.log('Images have varying dimensions:');
        uniqueDims.forEach(dim => {
          const count = dimensions.filter(d => d === dim).length;
          console.log(`  ${count} images are ${dim}`);
        });
      }
    });
  } catch (e) {
    console.error('Error checking image sizes:', e);
    process.exit(1);
  }
})(); 

File: scripts/check_sizes.py
#!/usr/bin/env python3
import os
from glob import glob
from PIL import Image

def main():
    script_dir = os.path.dirname(os.path.abspath(__file__))
    images_dir = os.path.join(script_dir, '..', 'public', 'img')
    patterns = ['*.jpg', '*.JPG', '*.jpeg', '*.JPEG', '*.png', '*.PNG']
    files = []
    for pat in patterns:
        files.extend(glob(os.path.join(images_dir, pat)))

    if not files:
        print('No images found.')
        return

    sizes = []
    for f in files:
        with Image.open(f) as img:
            sizes.append(img.size)  # (width, height)

    unique_sizes = set(sizes)
    if len(unique_sizes) == 1:
        w, h = unique_sizes.pop()
        print(f"All {len(files)} images are {w}x{h}")
    else:
        print('Images have varying dimensions:')
        counts = {size: sizes.count(size) for size in unique_sizes}
        for (w, h), count in counts.items():
            print(f"  {count} images are {w}x{h}")

if __name__ == '__main__':
    main() 